ztp/siteconfig-generator/siteConfig/siteConfigBuilder_test.go

```diff
@@ -225,3 +264,59 @@ func Test_getExtraManifestTemplatedRoles(t *testing.T) {
 		}
 	}
 }
+
+func Test_SNOClusterSiteConfigBuild(t *testing.T) {
+	sc := SiteConfig{}
+	err := yaml.Unmarshal([]byte(siteConfigTest), &sc)
+	assert.Nil(t, err)
+	CheckSiteConfigBuild(t, sc)
+}
+
+func Test_StandardClusterSiteConfigBuild(t *testing.T) {
+	sc := SiteConfig{}
+	err := yaml.Unmarshal([]byte(siteConfigStandardClusterTest), &sc)
+	assert.Nil(t, err)
+	CheckSiteConfigBuild(t, sc)
+}
+
+func CheckSiteConfigBuild(t *testing.T, sc SiteConfig) {
+	scBuilder, _ := NewSiteConfigBuilder()
+	scBuilder.SetLocalExtraManifestPath("../../source-crs/extra-manifest")
+	clustersCRs, err := scBuilder.Build(sc)
+	assert.Nil(t, err)
+	assert.Equal(t, len(clustersCRs), len(sc.Spec.Clusters))
+
+	for clusterId, cluster := range sc.Spec.Clusters {
+		testClusterCRs := clustersCRs[sc.Metadata.Name+"/"+cluster.ClusterName]
+		assert.NotNil(t, testClusterCRs)
+		expectedCRs := make([]interface{}, 0)
+		for _, sourceCR := range scBuilder.SourceClusterCRs {
+			mapSourceCR := sourceCR.(map[string]interface{})
+			assert.NotNil(t, mapSourceCR)
+			if mapSourceCR["kind"] == "BareMetalHost" || mapSourceCR["kind"] == "NMStateConfig" {
+				for nodeId, _ := range cluster.Nodes {
+					expectedCR, err := scBuilder.getClusterCR(clusterId, sc, mapSourceCR, nodeId)
```

Comment:
I'm dubious about this "call scBuilder to test scBuilder" methodology - What exactly are we trying to test here?

Reply:
I added the expected output from those test siteconfigs in order to make sure that whatever changes happen in the future can be validated for the same given siteconfigs.

Reply:
I see that the outer tests seem to be doing useful things, but here's what I see in this function:
1. we call `scBuilder.Build` to make all the cluster CRs
2. For each cluster we call `scBuilder.getClusterCr`, basically duplications what the `scBuilder.Build` did
3. We check that the test logic around  `scBuilder.getClusterCr` made the same thing that `scBuilder.Build` made
4. We marshal the result and that's what the caller checks against.

I think the check you mentioned on the expected SiteConfig relies on doing #1 and #4, but I don't think we need #2 or #3.  They just seem to be checking that our `scBuilder.Build` matches our local duplications of the code in this function.  Am I missing something?

Reply:
Steps 2 and 3 for testing the business logic changes in `scBuilder.Build` AND `scBuilder.getClusterCrs`. if there is a change in the func the unitTest will fail before comparing the end result output.  [L336](https://github.com/openshift-kni/cnf-features-deploy/pull/842/files#diff-debded0a9a2b5f548423222493ec145e61e597d7fc5df62c4bbd3320798351c1R336) . What do u think I can remove it and just check for the end output ?

Reply:
moved getClusterCR in another unitTest